pkg/apis/v2beta2/common.go

```diff
@@ -0,0 +1,115 @@
+	// +optional
+	MatchExpressions []LabelSelectorRequirement `json:"matchExpressions,omitempty" protobuf:"bytes,2,rep,name=matchExpressions"`
+}
+
+type LabelSelectorRequirement struct {
+	// key is the label key that the selector applies to.
+	// +patchMergeKey=key
+	// +patchStrategy=merge
+	Key string `json:"key" patchStrategy:"merge" patchMergeKey:"key" protobuf:"bytes,1,opt,name=key"`
+	// operator represents a key's relationship to a set of values.
+	// Valid operators are In, NotIn, Exists and DoesNotExist.
+	Operator LabelSelectorOperator `json:"operator" protobuf:"bytes,2,opt,name=operator,casttype=LabelSelectorOperator"`
+	// values is an array of string values. If the operator is In or NotIn,
+	// the values array must be non-empty. If the operator is Exists or DoesNotExist,
+	// the values array must be empty. This array is replaced during a strategic
+	// merge patch.
+	// +optional
+	Values     []string `json:"values,omitempty" protobuf:"bytes,3,rep,name=values"`
+	RegexValue string   `json:"regexValues,omitempty" protobuf:"bytes,3,rep,name=regexValues"`
+}
+
+// A label selector operator is the set of operators that can be used in a selector requirement.
+type LabelSelectorOperator string
+
+const (
+	LabelSelectorOpIn           LabelSelectorOperator = "In"
+	LabelSelectorOpNotIn        LabelSelectorOperator = "NotIn"
+	LabelSelectorOpExists       LabelSelectorOperator = "Exists"
+	LabelSelectorOpDoesNotExist LabelSelectorOperator = "DoesNotExist"
+	LabelSelectorOpMatch        LabelSelectorOperator = "Match"
+)
+
+func (ls *LabelSelector) Match(label map[string]string) (bool, error) {
+	var selector *metav1.LabelSelector
+	selector.MatchLabels = ls.MatchLabels
+
+	for _, requirement := range ls.MatchExpressions {
+		if requirement.Operator != LabelSelectorOpMatch {
+			selector.MatchExpressions = append(selector.MatchExpressions, metav1.LabelSelectorRequirement{
+				Key:      requirement.Key,
+				Operator: metav1.LabelSelectorOperator(requirement.Operator),
+				Values:   requirement.Values,
+			})
+		} else {
+			if v, ok := label[requirement.Key]; ok {
+				match, err := regexp.MatchString(requirement.RegexValue, v)
+				if err != nil || !match {
+					return false, err
+				}
+			}
+		}
+	}
+
+	sl, err := metav1.LabelSelectorAsSelector(selector)
+	if err != nil {
+		return false, err
+	}
+
+	return sl.Matches(labels.Set(label)), nil
+}
+
+func (ls *LabelSelector) Validate() error {
+	var selector *metav1.LabelSelector
+	selector.MatchLabels = ls.MatchLabels
+
+	for _, requirement := range ls.MatchExpressions {
+		if requirement.Operator != LabelSelectorOpMatch {
+			selector.MatchExpressions = append(selector.MatchExpressions, metav1.LabelSelectorRequirement{
+				Key:      requirement.Key,
+				Operator: metav1.LabelSelectorOperator(requirement.Operator),
+				Values:   requirement.Values,
+			})
+		} else {
+			_, err := regexp.Compile(requirement.RegexValue)
+			if err != nil {
+				return err
```

Comment:
Log an error here to tell the user why the regex is not working

Reply:
There is unified error handling, no need to print logs here.